### ISSUE TITLE
Question about Origin of dX3

### ISSUE THREAD

**Issue by Kaleye** [guilhermemd11/platypus#2]
Hello, 
I've been wondering where dX3 comes from. From what I can tell it calculates the flow rate of nitrous through the feed pipe. Is this correct? As such why is it used to calculate the vapor flow? 

Wouldn't the smallest area of the system (which should be the injector unless pipe area<injector), determine the mass flow rate(Ignoring losses due to friction)?

Cheers, 
Kaleye

**Comment by guilhermemd11** [comment]
Hi Kaleye,

The dX3 term computes the mass flow rate of vapour nitrous oxide only. 

Initially, when there is still liquid nitrous oxide inside the oxidiser tank, dX3 computes the mass flow rate of vapour nitrous oxide through the venting system. 

When all liquid nitrous oxide has been consumed, dX3 computes the mass flow rate of vapour nitrous oxide through both the venting system and the injector.

Please let me know if this answers your question.

Best regards,

Guilherme

**Comment by Kaleye** [comment]
So by venting system you mean whatever purge/vent system there might be on the rocket?

**Comment by guilhermemd11** [comment]
Yes. In particular, on the rocket motor, as some hybrid motors have a venting system to prevent the nitrous oxide from going supercritical.
